gammapy/modeling/models/cube.py

```diff
@@ -73,6 +74,10 @@ def __init__(
         self.apply_irf = {"exposure": True, "psf": True, "edisp": True}
         if apply_irf is not None:
             self.apply_irf.update(apply_irf)
+        if datasets_names is not None:
```

Comment:
As the default is an immutable str (`"all"`), you can just use it directly as a default, like `__init__(..., datasets_names="all")`.